infogami/utils/markdown/markdown.py

```diff
@@ -248,10 +248,11 @@ def unlink(self) :
         self.childNodes = None
 
     def setAttribute(self, attr, value) :
-        if not attr in self.attributes :
-            self.attributes.append(attr)
+        if attr in self.attributes :
+            self.attribute_values[attr] = value
 
-        self.attribute_values[attr] = value
+        
+        self.attributes.append(attr)
```

Comment:
The behavior is completely changed. We'll now hav duplicate vlaues in the `attributes` when setAttribute is called multiple times with the same attr.

Reply:
I have made the appropriate changes